src/AlertsManagement/AlertsManagement/ActionRuleCommands/GetAzureActionRule.cs

```diff
@@ -0,0 +1,287 @@
+
+        /// <summary>
+        /// Action rule name
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   ParameterSetName = ActionRuleByNameParameterSet,
+                   HelpMessage = "Name of action rule.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesParameterSet,
+                   HelpMessage = "Filter on Name of action rule.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesByTargetResourceIdParameterSet,
+                   HelpMessage = "Filter on Name of action rule.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesByTargetResourceGroupParameterSet,
+                   HelpMessage = "Filter on Name of action rule.")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Resource Group Name
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   ParameterSetName = ActionRuleByNameParameterSet,
+                   HelpMessage = "Resource Group Name in which action rule resides.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesParameterSet,
+                   HelpMessage = "Filter on Resource Group Name in which action rule resides.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesByTargetResourceIdParameterSet,
+                   HelpMessage = "Filter on Resource Group Name in which action rule resides.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesByTargetResourceGroupParameterSet,
+                   HelpMessage = "Filter on Resource Group Name in which action rule resides.")]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Resource Id
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesByTargetResourceIdParameterSet,
+                   HelpMessage = "Filter on Resource Id of the target resource of alert.")]
+        public string TargetResourceId { get; set; }
+
+        /// <summary>
+        /// Resource Type
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesParameterSet,
+                   HelpMessage = "Filter on Resource type of the target resource of alert.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesByTargetResourceIdParameterSet,
+                   HelpMessage = "Filter on Resource type of the target resource of alert.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = ListActionRulesByTargetResourceGroupParameterSet,
+                   HelpMessage = "Filter on Resource type of the target resource of alert.")]
+        [ResourceTypeCompleter]
+        public string TargetResourceType { get; set; }
```

Comment:
Makes sense. But as mentioned earlier, our api does not deal with mutually exclusive fields. It exposes all possible filters and in case of any conflict - no result is returned. For now, removed this parameter from 'ListActionRulesByTargetResourceIdParameterSet'.

Reply:
TargetResourceGroup and TargetResourceType are not mutually exclusive fields. The user may want to filter on particular resource group and further filter by resource type say only virtual machines.